tests/host_tools/cargo_build.py

```diff
@@ -70,3 +70,11 @@ def get_firecracker_binaries():
     jailer_bin_path = "{}/{}".format(out_dir, JAILER_BINARY_NAME)
 
     return fc_bin_path, jailer_bin_path
+
+
+def get_rustflags():
+    """Get the relevant rustflags for building/unit testing."""
+    rustflags = "-D warnings"
+    if platform.machine() == "aarch64":
+        rustflags += " -C link-arg=-lgcc -C link-arg=-lfdt "
```

Comment:
If I understand correctly, we would link against `-lgcc` and `-lfdt` for the firecracker binary used inside the integration tests and when cargo testing on arm. Why is this needed now, when we disallow warnings?

Reply:
We need this flag for aarch64 binary to link (these flags are always used when doing `cargo build` on aarch64. Take a look at the cargo/.config for arm64. Unfortunately there is no way (or I do not know any) of getting the flags set in cargo/.config for some platform from our python tests.

Reply:
Ok, so when compiling with `cargo build`, using `TARGET_CC=musl-gcc`, this has to link on aarch64 against `-lgcc` and `-lfdt` always (because musl-gcc lacks some symbols used in firecracker), and these flags can be found in `$HOME/.cargo/config`. I've looked on my x86 machine, and found this config file inside the .cargo dir inside the project, as well as this part responsible for aarch64 specific linking flags.

It makes sense, once we compile using rustc, and not `cargo build`, to specify these flags in the command line.

Reply:
Oh, actually, looked once more, and found out that we actually use `cargo build` to build the firecracker binaries. Using a .cargo/config with this directive, inside the project where this build happens, should be enough:

[target.'cfg(any(target_arch="arm", target_arch="aarch64"))']
rustflags = [
        "-C", "link-arg=-lgcc",
        "-C", "link-arg=-lfdt",
]

However, this is more like a nit now. I do not have a strong preference, to use this or what you've already pushed in this PR.